src/main/java/connectionpool/ConnectionPool.java

```diff
@@ -0,0 +1,398 @@
+package connectionpool;
+
+
+import java.sql.*;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Executor;
+
+public final class ConnectionPool {
+    private final static ConnectionPool instance = new ConnectionPool();
+    private BlockingQueue<Connection> connectionQueue;
+    private BlockingQueue<Connection> givenAwayConQueue;
+
+    private String driverName;
+    private String url;
+    private String user;
+    private String password;
+    private int poolSize;
+
+    public static ConnectionPool getInstance(){
+        return instance;
+    }
+
+    private ConnectionPool() {
+        DBResourceManager dbResourseManager = DBResourceManager.getInstance();
+        this.driverName = dbResourseManager.getValue(DBParameter.DB_DRIVER);
+
+        this.url = dbResourseManager.getValue(DBParameter.DB_URL);
+        this.user = dbResourseManager.getValue(DBParameter.DB_USER);
+        this.password = dbResourseManager.getValue(DBParameter.DB_PASSWORD);
+        try {
+            this.poolSize = Integer.parseInt(dbResourseManager
+                    .getValue(DBParameter.DB_POLL_SIZE));
+        } catch (NumberFormatException e) {
+            poolSize = 5;
+        }
+    }
+    public void initPoolData() throws ConnectionPoolException {
```

Comment:
you have to exec this method befor you take connection, so add it to constructor (it's the simpliest way to fix it)